x/upgrade/abci_test.go

```diff
@@ -410,3 +410,70 @@ func TestDumpUpgradeInfoToFile(t *testing.T) {
 	err = os.Remove(upgradeInfoFilePath)
 	require.Nil(err)
 }
+
+// TODO: add testcase to for `no upgrade handler is present for last applied upgrade`.
```

Comment:
how about this TODO?

Reply:
It is hard to add tests for a few cases without exposing/adding few functions. And we cannot move those tests to internal package (getting import cycles).